### DISCUSSION TITLE
BStats integration

### DISCUSSION THREAD

**Discussion by Andre601** [modrinth/.github#27] [Feature requests]
### Describe the solution you'd like

Option to add a project id from bStats to display specific stats in the project.

Ideally should multiple bStats projects be supported (in case your project supports multiple platforms) and options to define what stats to show (not just servers and players).

When this would be added, should perhaps own graphs be used, to allow them to fit better into the used site theme.

### Describe alternatives you've considered

Integrate the images, which may work, but look bad, especially on dark theme due to the white background.

### Additional context

bStats has a REST API that can be used to obtain statistics: https://bstats.org/help/rest-api

**Comment by intergrav** [comment]
Concept I made as it should probably also show that it uses bstats 
![image](https://user-images.githubusercontent.com/42325132/184937175-929685cf-3913-47ca-8992-93d2fc1c21fb.png)

**Comment by Arbee4ever** [comment]
If this was added, also add "Servercount" to the "Sort by" dropdown.

**Comment by ghost** [comment]
This is kinda out of scope for knossos itself to do, you could just put an updating image into your project body, it works fine. not gonna go on any further

**Comment by Andre601** [comment]
> This is kinda out of scope for knossos itself to do, you could just put an updating image into your project body, it works fine. not gonna go on any further

Not sure where this is "out of scope" here... In what way?

The only "updating image" I know of is the one bStats offers. And that one has many flaws and limitations in it:

- Image can only display number of servers and players. Nothing more, nothing less.
- Image has white background and no options for transparency, which makes it look out of place when looked at in dark theme.

Some people may not want to just display the servers a project is on, but other, custom stats they have and that would be something Modrinth could implement in some way.

If not for the frontend, this would at least be a thing to consider for the backend of Modrinth...

**Comment by ghost** [comment]
It could be hosted as a separate service making use of the bstats api to generate more customizable images.
